File: POO/Java/Abstração.java
package POO.Java;

public class Abstração {
    abstract class Veiculo {
        String modelo;
    
        Veiculo(String modelo) {
            this.modelo = modelo;
        }
    
        // Método abstrato: cada veículo vai implementar de forma diferente
        abstract void mover();
    }
    
    // Classe concreta implementa os detalhes
    class Carro extends Veiculo {
        Carro(String modelo) {
            super(modelo);
        }
    
        // Implementação do comportamento mover()
        @Override
        void mover() {
            System.out.println(modelo + " está dirigindo na estrada.");
        }
    }
    
}

// Explicação: Veiculo é uma abstração genérica. 
//  O usuário usa mover() sem saber os detalhes 
//  de como um carro se move internamente.

File: POO/Python/Abstração.py
from abc import ABC, abstractmethod

# Classe abstrata (modelo genérico)
class Veiculo(ABC):
    def __init__(self, modelo):
        self.modelo = modelo

    @abstractmethod
    def mover(self):
        pass

# Classe concreta com implementação específica
class Carro(Veiculo):
    def mover(self):
        print(f"{self.modelo} está dirigindo na estrada.")

# Explicação: A classe Veiculo define um 
#   método abstrato mover() que cada veículo 
#   deve implementar. Carro implementa esse comportamento 
#   específico.